### PR TITLE
SpingBootWeb

### PR DESCRIPTION
**Pull request by mykolaskrypka**
added dto's and mapper

### CODE REVIEW

File: src/main/java/mate/academy/repository/BookDaoImpl.java [hunk 1]
```diff
@@ -46,7 +44,19 @@ public List<Book> findAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery("SELECT b from Book b", Book.class).getResultList();
         } catch (Exception e) {
-            throw new RuntimeException("Can't get all books from db", e);
+            throw new EntityNotFoundException("Can't get all books from db", e);
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/repository/BookDaoImpl.java:47]: Check Exception message

File: src/main/java/mate/academy/repository/BookDaoImpl.java [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
     private final SessionFactory sessionFactory;
 
-    @Autowired
-    public BookDaoImpl(SessionFactory sessionFactory) {
-        this.sessionFactory = sessionFactory;
-    }
-
     @Override
     public Book save(Book book) {
         Transaction transaction = null;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/repository/BookDaoImpl.java:21]: For what this?

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package mate.academy.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java:7]: Why you choose this naming of entity?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java (hunk 3 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/dto/CreateBookRequestDto.java:7]: spWeb

File: src/main/java/mate/academy/repository/BookDaoImpl.java (hunk 2 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/repository/BookDaoImpl.java:21]: this part is missing in the current version
spWeb2

File: src/main/java/mate/academy/repository/BookDaoImpl.java (hunk 1 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/repository/BookDaoImpl.java:47]: what do you mean?

File: src/main/java/mate/academy/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping
+    public void createBook(@RequestBody CreateBookRequestDto requestDto) {
+        bookService.save(requestDto);
+    }
+
```
Comment (koval-kn) [inline src/main/java/mate/academy/controller/BookController.java:35]: redundant blank line

**Review by koval-kn — APPROVED** [review]

**Pull request merged**